lib/select.c

```diff
@@ -219,6 +219,8 @@ int Curl_select(curl_socket_t maxfd,   /* highest socket number */
 #else
   r = select((int)maxfd + 1, fds_read, fds_write, fds_err, ptimeout);
 #endif
+  if(r < 0)
+    r = -1;
```

Comment:
The only documented negative value select() can return is -1. This seems superfluous.

Reply:
I wanted to align it here with the way the poll return code is handled. On Windows select() is defined to return SOCKET_ERROR on error, which is indeed currently defined to -1.

Reply:
"currently" ? select() is defined to work like this in POSIX and it **will not** be changed by anyone as that would break a bazillion apps. And if someone would provide a crazy broken select() and some point that can return something else then we deal with that then and for that particular broken system.

Reply:
Okay, I understand, but then we can also drop that same if condition below, see my next comment.

Reply:
Now part of https://github.com/curl/curl/pull/5880.